consumer.go

```diff
@@ -47,33 +48,48 @@ type ConsumerConfig struct {
 	WriteTimeout time.Duration
 }
 
-func StartConsumer(config ConsumerConfig) (c *Consumer, err error) {
-	if len(config.Topic) == 0 {
-		err = errors.New("creating a new consumer requires a non-empty topic")
-		return
+// validate ensures that this configuration is well-formed.
+func (c *ConsumerConfig) validate() error {
+	if len(c.Topic) == 0 {
+		return errors.New("creating a new consumer requires a non-empty topic")
 	}
 
-	if len(config.Channel) == 0 {
-		err = errors.New("creating a new consumer requires a non-empty channel")
-		return
+	if len(c.Channel) == 0 {
+		return errors.New("creating a new consumer requires a non-empty channel")
+	}
+
+	return nil
+}
+
+// defaults will set up this configuration with the global defaults where they
+// were not already set.
+func (c *ConsumerConfig) defaults() {
+	if c.MaxInFlight == 0 {
+		c.MaxInFlight = DefaultMaxInFlight
 	}
 
-	if config.MaxInFlight == 0 {
-		config.MaxInFlight = DefaultMaxInFlight
+	if c.DialTimeout == 0 {
+		c.DialTimeout = DefaultDialTimeout
 	}
 
-	if config.DialTimeout == 0 {
-		config.DialTimeout = DefaultDialTimeout
+	if c.ReadTimeout == 0 {
+		c.ReadTimeout = DefaultReadTimeout
 	}
 
-	if config.ReadTimeout == 0 {
-		config.ReadTimeout = DefaultReadTimeout
+	if c.WriteTimeout == 0 {
+		c.WriteTimeout = DefaultWriteTimeout
 	}
+}
 
-	if config.WriteTimeout == 0 {
-		config.WriteTimeout = DefaultWriteTimeout
+// NewConsumer configures a new consumer instance.
+func NewConsumer(config ConsumerConfig) (c *Consumer, err error) {
+	if configError := config.validate(); configError != nil {
+		err = configError
```

Comment:
Seems like this can be shorten:
```go
if err = config.validate9); err != nil {
    return
}
```
(unless there's a reason to express it this way)